Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2024 Adam Topaz All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Monoidal.OfChosenFiniteProducts.Symmetric
+import Mathlib.CategoryTheory.Limits.Constructions.FiniteProductsOfBinaryProducts
+
+/-!
+
+# Categories with chosen finite products
+
+We introduce a class, `ChosenFiniteProducts`, which bundles explicit choices
+for a terminal object and binary products in a category `C`.
+This is primarily useful for categories which have finite products with good
+definitional properties, such as the category of types.
+
+Given a category with such an instance, we also provide the associated
+symmetric monoidal structure so that one can write `X ⊗ Y` for the explicit
+binary product and `𝟙_ C` for the explicit terminal object.
+
+-/
+
+namespace CategoryTheory
+
+universe v u
+
+/--
+An instance of `ChosenFiniteProducts C` bundles an explicit choice of a binary
+product of two objects of `C`, and a terminal object in `C`.
+
+Users should use the monoidal notation: `X ⊗ Y` for the product and `𝟙_ C` for
+the terminal object.
+-/
+class ChosenFiniteProducts (C : Type u) [Category.{v} C] where
```

Comment:
We can probably live with this, but I do think that the name is a bit of a lie, because neither does this class give you any way to talk about finite products, nor does it actually imply the existence of a good choice of finite products.

Reply:
Yes, I agree, but I think the more "correct" name (e.g. "ChosenTerminalObjectsAndBinaryProducts") is too long. I think "ChosenFiniteProducts" is a good compromise, given existing names like "monoidalOfChosenFiniteProducts".